addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightDiscover.java

```diff
@@ -77,61 +86,59 @@ public MilightDiscover(DiscoverResult discoverResult, int resendTimeoutInMillis,
     }
 
     public void dispose() {
+        stopResend();
         willbeclosed = true;
         datagramSocket.close();
         try {
             join(500);
         } catch (InterruptedException e) {
         }
         interrupt();
-        if (datagramSocket != null) {
```

Comment:
I miss destructors in java. I made it a habit to use a dispose method instead, but i tend to forget that an API consumer can call this multiple times in contrast to a destructor. I guarded the dispose call correctly now.